app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+require_relative '../data_mapper_setup'
+require 'dm-validations'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  property :id, Serial
+
+  property :email, String, required: true, unique: true,
+    format: :email_address
+
+  property :password_digest, Text
+    validates_confirmation_of :password
+    validates_presence_of :password
+
+  property :name, String, required: true
+
+  property :user_name, String, required: true, unique: true
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+    user = first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
+      user
+    else
+      nil
+    end
+  end
+end
```

Comment:
no relationship between user and peeps? How would you get all the peeps that belong to a user?